templates/tapestry-d3-vue/src/components/NodeModal.vue

```diff
@@ -0,0 +1,242 @@
+<template>
+  <div id="node-modal-container">
+    <div
+      class="modal fade"
+      id="createNewNodeModal"
+      tabindex="-1"
+      role="dialog"
+      aria-labelledby="createNewNodeModalLabel"
+      aria-hidden="true"
+    >
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h4 class="modal-title" id="createNewNodeModalLabel"></h4>
+          </div>
+          <div class="modal-body" id="createNewNodeModalBody">
+            <div id="add-node-error-msg"></div>
+            <form class="form-container">
+              <label>
+                Title
+                <input
+                  id="add-node-title-input"
+                  name="title"
+                  type="text"
+                  placeholder="Enter the topic title"
+                  required
+                />
+              </label>
+              <label>
+                Thumbnail
+                <input
+                  id="add-node-thumbnail-input"
+                  name="imageURL"
+                  type="url"
+                  placeholder="Enter the URL for the thumbnail"
+                  required
+                />
+              </label>
+              <label>
+                Media Type
+                <div class="dropdown">
+                  <select id="mediaType" name="mediaType">
+                    <option value="default">Select type:</option>
+                    <option value="video" selected>Video</option>
+                    <!--<option value="image">image</option>-->
+                  </select>
+                </div>
+              </label>
+              <label>
+                Media Format
+                <div class="dropdown">
+                  <select id="mediaFormat" name="mediaFormat" v-model="selectedMediaFormat">
+                    <option value>Select format:</option>
+                    <option value="mp4">MP4</option>
+                    <option value="h5p">H5P</option>
+                    <!--<option value="jpeg">JPEG</option>-->
+                  </select>
+                </div>
+              </label>
+              <div
+                id="contents-details"
+                class="content-details"
+                style="display: none"
+                v-show="showContentDetails"
+              >
+                <h3>Content Details</h3>
+                <div id="mp4-content" class="mp4-content" v-show="showMp4Content">
```

Comment:
Instead of "showMp4Content", please use actual logic (e.g. formData.mediaType == "mp4" or something like that after you bind all your form elements to models). This comment applies similarly to the rest of your code as well.